tests/test_aurora.py

```diff
@@ -0,0 +1,159 @@
+import numpy as np
+import torch
+from graph_weather.models.aurora import (
+    IntegrationLayer,
+    GenCastConfig,
+    Fengwu_GHRConfig,
+    Swin3DEncoder,
+    Decoder3D,
+    LoraLayer,
+    PerceiverProcessor
+)
+
+def test_aurora_integration():
+    """
+    Test the Aurora pipeline by integrating all components in a single test case.
+    """
+    # Set device
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
These tests should be done on small enough inputs and such to not really matter if there is a GPU. To make this simpler, I would suggest removing this line and device calls in general in the tests, they should all be able to run on a CPU fast enough.
```suggestion
```